02-microarray/ortholog-mapping_microarray_01_ensembl.Rmd

```diff
@@ -0,0 +1,355 @@
+
+refine.bio data uses Ensembl gene identifiers, which will be in the first column.
+
+```{r}
+# Let's take a look at the first 6 items in the first column of `df`
+head(df[, 1])
+```
+
+In zebrafish, Ensembl gene identifiers begin with `ENSDARG`.
+
+Now let's do the conversion!
+We'll use the `zebrafish` data from `hcop`; this is what `zebrafish` looks like.
+
+```{r}
+zebrafish
+```
+
+We're interested in the `human_symbol` and `zebrafish_ensembl` columns specifically.
+
+```{r}
+human_symbol_df <- zebrafish %>%
+  # we'll want to subset zebrafish to only the columns we're interested in
+  dplyr::select(zebrafish_ensembl, human_symbol) %>%
+  # only retain the genes from the expression matrix that are in zebrafish data
+  dplyr::inner_join(df, by = c("zebrafish_ensembl" = "Gene"))
+```
+
+Here's what the new data frame looks like:
+
+```{r}
+head(human_symbol_df, 25)
+```
+
+Let's get a summary of the gene symbols returned in the `human_symbol` column of our mapped data frame.
+
+```{r}
+# We can use the `summary()` function to get a better idea of the distribution of symbols in the `Symbol` column
+summary(as.factor(human_symbol_df$human_symbol)) # We need to use the `as.factor()` function here in order to get the count of each unique value returned
+```
+
+Looks like we have mapped symbols!
+It does not explicitly tell us if we have IDs that did not map to symbols, which would result in NAs so let's explore to find out if we do have any.
+
+```{r}
+any(is.na(human_symbol_df$human_symbol)) # Logical statement to check if there are any NAs in the `human_symbol` column of our mapped data frame
+```
+
+There are no NAs!
+This means that all of our zebrafish Ensembl IDs mapped to at least one human gene symbol!
+This is exciting because it is not always the case everything maps across species and that is okay.
+
+However, if you have almost all NAs it is possible that the function was executed incorrectly or you may want to consider using a different gene identifier, if possible.
+
+## Write results to file
+
+Now let's write our results to file.
+
+```{r}
+readr::write_tsv(
+  human_symbol_df,
+  file.path(
+    results_dir,
+    "GSE71270_ensembl_to_human_gene_symbol.tsv" # Replace with a relevant output file name
+  )
+)
+```
+
+*Some things to note:*
+
+* Multiple zebrafish Ensembl gene IDs map to the same human symbol which results in duplicated human gene symbols.
+Some downstream tools may need this to be resolved.
+One way to resolve this may be to map to a different gene identifier.
+See our [gene ID annotation microarray notebook](https://alexslemonade.github.io/refinebio-examples/02-microarrary/gene-id-annotation_microarray_01_ensembl.html) for an example of mapping between gene identifiers.
+* If a zebrafish Ensembl gene ID maps to multiple human symbols, the gene expression values get duplicated.
+Let's look at the `ENSDARG00000069142` example below.
+
+```{r}
+human_symbol_df %>%
```

Comment:
I like this using an example to illustrate the idea you are explaining can you do the same for the first bullet point as well? 

I think it makes sense to split this into two sections, something like: `collapsing duplicate human IDs` and `collapsing/summarizing multiple zebrafish IDs`. 

But of course the `collapsing/summarizing multiple zebrafish IDs` strategy we will have to think about what makes the most sense. This will be dependent on what a user would be using this for downstream, which we can't know for certainty, but we can try to model what we think the most common use cases would be.